src/drivers/imx/sai.c

```diff
@@ -413,6 +370,93 @@ static int sai_get_hw_params(struct dai *dai,
 	return 0;
 }
 
+static int sai_get_word_width(enum sof_ipc_frame fmt)
+{
+	switch (fmt) {
+	case SOF_IPC_FRAME_S16_LE:
+		return 16;
+	case SOF_IPC_FRAME_S24_4LE:
+	case SOF_IPC_FRAME_S32_LE:
+		return 32;
+	default:
+		return -EINVAL;
+	}
+}
```

Comment:
I think there is already a common function that does this conversion.

Reply:
Yes, "get_sample_bytes" returns the number of bytes for a format, will replace it in the following push.